v-next/hardhat/src/internal/builtin-plugins/coverage/coverage-manager.ts

```diff
@@ -557,4 +568,54 @@ export class CoverageManagerImplementation implements CoverageManager {
 
     return formatTable(rows);
   }
+
+  async #writeHtmlReport(
+    report: Report,
+    htmlReportPath: string,
+  ): Promise<void> {
+    const baseDir = process.cwd();
+    const coverageMap = libCoverage.createCoverageMap({});
+
+    // Construct coverage data for each tested file,
+    // detailing whether each line was executed or not.
+    for (const [p, fileCoverageInput] of Object.entries(report)) {
+      const testedFilePath = path.isAbsolute(p) ? p : path.join(baseDir, p);
```

Comment:
I wouldn't block on this, but: I am surprised about this. We generate the report, so we should know if the keys are relative or absolute paths. (The `Report` interface names the key `relativePath`, so I guess they are meant to be relative).

Another concern is that, if they are relative, it's unclear if they are relative to the cwd or the project's root (IMO it should be the latter).

Again: I wouldn't do anything on this PR, but this seems like technical debt. Not sure how/if we are capturing those.